### PR TITLE
Junyu wang/poll question size

### PR DESCRIPTION
**Pull request by JessieWang0706**
## Overview
make poll header resizable, debug the poll question page
  ## Changes Made

  
1. the size of the poll question textview will automatically adjust to the length of text
2. added an textview to represent the length of the poll question out of /120 and it only appears once the user start to input
3. users cannot input more text after 120 character limit (#86 )
4. when users select draft the poll question size textview will also disappear because the length =0
5. the poll question page header will not shrunk when the listview is too long. 
6. The header became more properly constraint and the choices listview because scrollable with the max height 500 tested on 5-7 inches screens (#80 )
7. Adjust the Question Textview Layout: If the role is user, the questionTextView is default centered, If the role is admin, the questionTextView Layout will programmatically be aligned start
8. Added some bottom margin and adjust divider position for the poll header
9. Match Onboard view with new create Poll Fragment
  ## Test Coverage

  I manually test on emulator and my own phone, it all works as intended
  Devices Pixel 2XL API 29, Huawei Mate 20,

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/pollo/CreatePollFragment.kt [hunk 1]
```diff
@@ -122,6 +122,22 @@ class CreatePollFragment : Fragment(), DraftAdapter.DraftsDelegate, DraftAdapter
         groupMenuOptionsView.renameGroup.visibility = View.GONE
 
         groupMenuOptionsView.closeButton.setOnClickListener { dismissPopup() }
+
+        poll_question.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+                val questionLength=s?.length
+                    word_count.visibility=View.VISIBLE;
+                    word_count.text = "$questionLength"+"/120"}
```
Comment (Shiyuan-Huang-23) [inline app/src/main/java/com/cornellappdev/android/pollo/CreatePollFragment.kt:130]: Please format this section to have consistent indentation and consistent spacing when assigning values. Semi-colons are not needed. Also, the string can be `"$questionLength/120"`.  Please also put the ending curly braces on a separate line.

File: app/src/main/res/layout/poll_recyclerview_item_row.xml [hunk 2]
```diff
@@ -16,24 +16,34 @@
...
+                android:layout_height="55dp"
+                android:layout_marginStart="42dp"
+                android:layout_marginEnd="42dp"
                 android:layout_marginTop="1dp"
                 android:fontFamily="sans-serif"
+                android:gravity="center"
                 android:letterSpacing="0.03"
                 android:maxLines="3"
-                android:text="Favorite Ice Cream"
+                android:text="dhuewduwedhuewdhuewd"
```
Comment (Shiyuan-Huang-23) [inline app/src/main/res/layout/poll_recyclerview_item_row.xml:35]: We already talked about this, but please change this text.

**Review by Shiyuan-Huang-23 — APPROVED** [review]
Looks good overall. Please look at the comments I left regarding style and also autoformat `PollsRecyclerAdapter.kt` to remove semi-colons and add appropriate spaces when assigning variables. 

Also, when poking around in my emulator, I noticed that when adding a question with a lot of characters, the characters end up shrinking in size. Do you think you could increase the size of the header instead of having the font size change?

One last thing- the changes to `fragment_create_poll.xml` seem to be causing the onboarding to no longer line up with the underlying screen. You don't have to worry about fixing this issue, though. I can make the appropriate changes to onboarding in a later PR since I worked on onboarding last semester.

File: app/src/main/java/com/cornellappdev/android/pollo/CreatePollFragment.kt (hunk 1 above)
Comment (kevinsun-dev) [reply app/src/main/java/com/cornellappdev/android/pollo/CreatePollFragment.kt:130]: To that note, there's a auto-format shortcut in Android Studio `Ctrl+Alt+L`. Just spam that on the files when you commit and you'll probably be good.

File: app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsRecyclerAdapter.kt [hunk 3]
```diff
@@ -57,17 +61,21 @@ class PollsRecyclerAdapter(private var polls: ArrayList<Poll>,
...
-            val headerHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 100f, displayMetrics).toInt()
+            val headerHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 140f, displayMetrics).toInt()
             val cellHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 55f, displayMetrics).toInt()
             val adminControlsHeight = if (role == User.Role.ADMIN) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 85f, displayMetrics).toInt() else 0
             // 86 dp is the height of header, 53dp is height of cell
             val tmpHeight = headerHeight + cellHeight * poll.answerChoices.count() + adminControlsHeight // 53 is cell height including top margin
+            height = if (tmpHeight <= 1500) tmpHeight else 1500
 
-            height = if (tmpHeight <= 1250) tmpHeight else 1250
+            recyclerviewHeight=cellHeight*poll.answerChoices.count();
```
Comment (kevinsun-dev) [inline app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsRecyclerAdapter.kt:71]: Yeah, note that we keep two spaces on either side of our operators for readability. ` * ` and ` = ` in this case. In addition, there's an extraneous semicolon at the end of the line.

**Review by kevinsun-dev — APPROVED** [review]
Overall, just run a quick formatter over all these files before merging into master. Well done @JessieWang0706!

**Review by Shiyuan-Huang-23 — APPROVED** [review]
Looks good to me!

**Pull request merged**